### PR TITLE
[arpack-ng] Add new port

### PR DESCRIPTION
**Pull request by FabienPean**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [x] The versioning scheme in `vcpkg.json` matches what upstream says. 
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See docs/examples/adding-an-explicit-usage.md for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

The PR is a draft and does not contain correct versioning scheme at the moment. I want to see if the results of the CI here and then request a new release upstream if no upstream changes needed, updated here afterwards when finalizing the PR. Only core package is integrated at the moment, all options are disabled. 

This is a Fortran package with plenty of quirks (msys/mingw related on Windows especially), so there are some things I must have missed, feedback appreciated !

Minimal compiling example: [arpack_mwe.zip](https://github.com/microsoft/vcpkg/files/10538846/arpack_mwe.zip)

### CODE REVIEW

**Comment by FabienPean** [comment]
Failure on `x64_uwp`: it relies on `clapack` which does not seem to offer CMake built-in targets `LAPACK::LAPACK` out of `find_package(LAPACK REQUIRED)`

This leads to the trigger of this part of the upstream CMakeLists

```cmake
    find_package(LAPACK REQUIRED)

    # LAPACK::LAPACK target was already created at this point by FindLAPACK.cmake if cmake version >= 3.18
    if (NOT TARGET LAPACK::LAPACK) # Create target "at hand" to ensure compatibility if cmake version < 3.18
        add_library(LAPACK::LAPACK INTERFACE IMPORTED)
        set_target_properties(LAPACK::LAPACK PROPERTIES INTERFACE_LINK_LIBRARIES "${LAPACK_LIBRARIES}")
    endif()
```

Is it something to fix upstream or vcpkg related ? Or simply disabling support for uwp

**Comment by BillyONeal** [comment]
I manually checked that the SHA in here is not affected by https://github.com/microsoft/vcpkg/issues/29288

**Comment by FabienPean** [comment]
Ready for review I guess. Here some comments I would particularly appreciate feedback for:

* uwp is skipped for now because of the problem mentioned in previous comment.
* I am not sure how to set the ci.baseline.txt file here or if it is necessary besides the `supports` clause in the manifest
* arm is passing on the CI but I am not convinced it is really supported (ci does not seem to build anything)
* The port installs several files with simple names in its own folder but they are still used as in `#include<arpack.hpp>` not sure if it conflicts with [this rule](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#unique-port-attribution-rule) 
* Does it require further upstream changes before an initial accept of the port ? If not, then I would request a new release to use the correct version scheme for the package.

**Comment by LilyWangLL** [comment]
I installed `arpack-ng:x64-windows` locally, there is a `libarpack.dll.a` in lib folder, is it correct?
image

**Comment by FabienPean** [comment]
Yes, the package is built through msys2 and mingw. For a point of comparison, lapack-reference generates similar things.

**Review by LilyWangLL — DISMISSED** [review]

File: ports/arpack-ng/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,34 @@
+include(vcpkg_find_fortran)
```
Comment (BillyONeal) [inline ports/arpack-ng/portfile.cmake:1]:
```suggestion
```

This should already be set for you.

**Review by BillyONeal — DISMISSED** [review]
This LGTM as soon as upstream does the release to get versioning to match.

File: ports/arpack-ng/portfile.cmake (hunk 1 above)
Comment (FabienPean) [reply ports/arpack-ng/portfile.cmake:1]: It seems needed actually. It failed without it, and other ports with Fortran also have it
https://github.com/microsoft/vcpkg/blob/703a8113f42e794534df5dfc5cece5dabcb949d0/ports/lapack-reference/portfile.cmake#L13

**Comment by FabienPean** [comment]
Sorry for potential notification overload, I thought it was possible to re-request review from both. Also, looks like the bot is KO

File: ports/arpack-ng/portfile.cmake (hunk 1 above)
Comment (LilyWangLL) [reply ports/arpack-ng/portfile.cmake:1]: Yes, without `include(vcpkg_find_fortran)`, the latest vcpkg will output the following error:
```
CMake Error at ports/arpack-ng/portfile.cmake:2 (vcpkg_find_fortran):
  Unknown CMake command "vcpkg_find_fortran".
```

**Comment by JavierMatosD** [comment]
Thanks!

**Comment by JackBoosY** [comment]
```
-- Performing post-build validation
warning: Detected outdated dynamic CRT in the following files:
    D:\packages\arpack-ng_x86-windows\debug\bin\libarpack.dll:msvcrt.dll
    D:\packages\arpack-ng_x86-windows\bin\libarpack.dll:msvcrt.dll
To inspect the dll files, use:
    dumpbin.exe /dependents mylibfile.dll
error: Found 1 post-build check problem(s). To submit these ports to curated catalogs, please first correct the portfile: C:\a\1\s\ports\arpack-ng\portfile.cmake
error: building arpack-ng:x86-windows failed with: POST_BUILD_CHECKS_FAILED
```

**Pull request merged**